src/parser/data/Variable.h

```diff
@@ -14,10 +14,19 @@ class Variable {
   std::string _name;
 
  public:
-  explicit Variable(std::string name) : _name{std::move(name)} {};
+  explicit Variable(std::string name) : _name{std::move(name)} {
+    // Verify variable name is not empty
+    AD_CHECK(_name.length() > 1);
+    // normalise notation for consistency
+    if (_name[0] == '$') {
+      _name[0] = '?';

```

Comment:
if you don't check that the first character is either $ or ? you can do the 
overwrite unconditionally. But also asserting this would probably be better.

Reply:
I think it's better to avoid the unecessary write operation if possible and let branch prediction skip this constructor in most cases